platform/default/mbgl/storage/offline_download.cpp

```diff
@@ -230,6 +230,7 @@ void OfflineDownload::ensureResource(const Resource& resource, std::function<voi
             && util::mapbox::isMapboxURL(resource.url)
             && offlineDatabase.offlineMapboxTileCountLimitExceeded()) {
             observer->mapboxTileCountLimitExceeded(offlineDatabase.getOfflineMapboxTileCountLimit());
+            setState(OfflineRegionDownloadState::Inactive);
```

Comment:
`setState(OfflineRegionDownloadState::Inactive)` triggers a `statusChanged` notification to the observer with the new `downloadState`. This is related to my comment here: https://github.com/mapbox/mapbox-gl-native/pull/4221#discussion_r55572448.